clang/test/CodeGenSYCL/union-kernel-param1.cpp

```diff
@@ -0,0 +1,83 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -I %S/Inputs -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+
+// This test checks if compiler accepts union as kernel parameters.
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+
+union MyUnion {
+  int FldInt;
+  int FldArr[3];
+};
+
+MyUnion GlobS;
+
+void test0() {
+  MyUnion S = GlobS;
+  MyUnion S0 = {0};
+  {
+    buffer<MyUnion, 1> Buf(&S0, range<1>(1));
+    queue myQueue;
+    myQueue.submit([&](handler &cgh) {
+      auto B = Buf.get_access<access::mode::write>(cgh);
+      cgh.single_task<class MyKernel>([=] {B; S; });
+    });
+  }
+}
+
+// CHECK MyKernel parameters
```

Comment:
@elizabethandrews  can you take a look at this?  The union should be a simple un-decomposed value.

Reply:
From the Sema tests below, it looks like parameters are being passed individually as well? This in incorrect right?

Reply:
Yes, the test seems incorrect. I will update the test. Thanks Erich and Elizabeth.